components/utility-components/redeem-button.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState, useEffect, useContext } from "react";
+import axios from "axios";
+import { Button, Spinner } from "@nextui-org/react";
+import { useTheme } from "next-themes";
+import { ProfileMapContext } from "../../utils/context/context";
+import { getLocalStorageData } from "../utility/nostr-helper-functions";
+import { SHOPSTRBUTTONCLASSNAMES } from "../utility/STATIC-VARIABLES";
+import { nip19 } from "nostr-tools";
+import { LightningAddress } from "@getalby/lightning-tools";
+import { CashuMint, CashuWallet, payLnInvoiceWithToken, getEncodedToken } from "@cashu/cashu-ts";
+import RedemptionModal from "./redemption-modal";
+import { formatWithCommas } from "./display-monetary-info";
+
+function decodeBase64ToJson(base64: string): any {
+  // Step 1: Decode the base64 string to a regular string
+  const decodedString = atob(base64);
+  // Step 2: Parse the decoded string as JSON
+  try {
+    const json = JSON.parse(decodedString);
+    return json;
+  } catch (error) {
+    console.error("Error parsing JSON from base64", error);
+    throw new Error("Invalid JSON format in base64 string.");
+  }
+}
+
+export default function RedeemButton({ token }: { token: string }) {
+  const [lnurl, setLnurl] = useState("");
+  const profileContext = useContext(ProfileMapContext);
+  const { npub, decryptedNpub, mints, relays } = getLocalStorageData();
+
+  const [openRedemptionModal, setOpenRedemptionModal] = useState(false);
+  const [isPaid, setIsPaid] = useState(false);
+  const [isCashu, setIsCashu] = useState(false);
+  const [isRedeeming, setIsRedeeming] = useState(false)
+  const [proofs, setProofs] = useState([]);
+  const [tokenAmount, setTokenAmount] = useState();
+  const [formattedTokenAmount, setFormattedTokenAmount] = useState();
+
+  const { theme, setTheme } = useTheme();
+
+  const [randomNpub, setRandomNpub] = useState<string>("");
+  const [randomNsec, setRandomNsec] = useState<string>("");
+
+  useEffect(() => {
+    axios({
+      method: "GET",
+      url: "/api/nostr/generate-keys",
+    })
+      .then((response) => {
+        setRandomNpub(response.data.npub);
+        setRandomNsec(response.data.nsec);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }, []);
+
+  useEffect(() => {
+    const decodedToken = decodeBase64ToJson(token);
+    const proofs = decodedToken.token[0].proofs;
+    setProofs(proofs);
+    const totalAmount = proofs.reduce(
```

Comment:
change this to 
```
const totalAmount =
      Array.isArray(proofs) && proofs.length > 0
        ? proofs.reduce((acc, current) => acc + current.amount, 0)
        : 0;
```
so the error does not occur when entering a chat with in properly formed cashu token

Reply:
Also perhaps you have to look into why this cashu token for 0 sats was sent.